docs/en/platform-engineers/cloud-services.md

```diff
@@ -11,104 +11,426 @@ The following practice could be considered:
   schematic:
     cue:
       template: |
         output: {
-        	apiVersion: "database.example.org/v1alpha1"
-        	kind:       "PostgreSQLInstance"
-        	metadata:
-        		name: context.name
+        	apiVersion: "database.alibaba.crossplane.io/v1alpha1"
+        	kind:       "RDSInstance"
         	spec: {
-        		parameters:
-        			storageGB: parameter.storage
-        		compositionSelector: {
-        			matchLabels:
-        				provider: parameter.provider
+        		forProvider: {
+        			engine:                parameter.engine
+        			engineVersion:         parameter.engineVersion
+        			dbInstanceClass:       parameter.instanceClass
+        			dbInstanceStorageInGB: 20
+        			securityIPList:        "0.0.0.0/0"
+        			masterUsername:        parameter.username
+        		}
+        		writeConnectionSecretToRef: {
+        			namespace: context.namespace
+        			name:      context.outputSecretName
         		}
-        		writeConnectionSecretToRef:
-        			name: parameter.secretname
+        		providerConfigRef: {
+        			name: "default"
+        		}
+        		deletionPolicy: "Delete"
+        	}
+        }
+        parameter: {
+        	engine:          *"mysql" | string
+        	engineVersion:   *"8.0" | string
+        	instanceClass:   *"rds.mysql.c1.large" | string
+        	username:        string
+        }
+```
+
+Noted: In application, application developers need to use property `outputSecretName` as the secret name which is used to store all connection
+items of cloud resource connections information.
+
+### Step 2: Prepare TraitDefinition `service-binding` to do env-secret mapping
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: TraitDefinition
+metadata:
+  annotations:
+    definition.oam.dev/description: "binding cloud resource secrets to pod env"
+  name: service-binding
+spec:
+  appliesToWorkloads:
+    - webservice
+    - worker
+  schematic:
+    cue:
+      template: |
+        patch: {
+        	spec: template: spec: {
+        		// +patchKey=name
+        		containers: [{
+        			name: context.name
+        			// +patchKey=name
+        			env: [
+        				for envName, v in parameter.envMappings {
+        					name: envName
+        					valueFrom: {
+        						secretKeyRef: {
+        							name: v.secret
+        							if v["key"] != _|_ {
+        								key: v.key
+        							}
+        							if v["key"] == _|_ {
+        								key: envName
+        							}
+        						}
+        					}
+        				},
+        			]
+        		}]
         	}
         }
 
         parameter: {
-        	secretname: *"db-conn" | string
-        	provider:   *"alibaba" | string
-        	storage:    *20 | int
+        	envMappings: [string]: [string]: string
         }
-EOF
 ```
 
-## Step 3: Verify
+Developers explicitly sets parameter `envMappings` to mapping all environment names with secret key. Here is an example.
+
+```yaml
+envMappings:
```

Comment:
write a full trait schema in Application:

```
...
		  traits:
			  - type: service-binding
				properties:
					envMappings:
						# environments refer to db-conn secret
						DB_PASSWORD:
							secret: db-conn
							key: password 
...
```

Reply:
I did in line https://github.com/oam-dev/kubevela/pull/1264/files#diff-1fa7e2b57330a2a24a633341a211de9329084ceb9bd7ea6918e112806aba4e5dR197-R208

Reply:
Here is an example to show how to write envMapping. As there is no other place for the introduction.

Reply:
Done.